Conditional reporting

lillysGame opened the pull request:
I would like to be able to limit what gets reported to jira. as it works now, my tests will fail and skip the rest, but also creates jira tickets for the skipped tests, maybe add a conditional to every field configuration - my main focus is if the specific field is going to be blank, don't report the ticket

Commit:
Fix escaping of JSON payload.

Use javax.json to create a JSON payload that escapes the content
that is sent to Jira.  This fixes 400 Bad request responses from
Jira whenever the test failure message has a double quote.

This also reformats the data sent to Jira so that it's formatted a bit
more nicely.

Commit:
Mark the workspace field as transient.

This fixes a crash when attempting to save a job that uses this plugin and
the job has been run on a slave: "Can't send a remote FilePath to a different
remote channel".

maplesteve committed:
Merge pull request #2 from tipbit/fix-json-escaping

Fix escaping of JSON payload.

maplesteve committed:
Merge pull request #3 from tipbit/fix-transient-workspace

Mark the workspace field as transient.

Commit:
Fix a NullPointerException when there are no test results.

If there are no test results (e.g. if the build fails before the tests run)
then build.getAction(AbstractTestResultAction.class) returns null.
Add a check for this case.

Commit:
Add job URL to the Jira ticket.

Add the URL for the job that caused the test failure to the Jira ticket.

maplesteve committed:
Merge pull request #4 from tipbit/fix-crash-missing-tests

Fix a NullPointerException when there are no test results.

maplesteve committed:
Merge pull request #5 from tipbit/add-job-url

Add job URL to the Jira ticket.

Commit:
fix for Jenkins 1.577+

Commit:
fix the compile error

maplesteve committed:
Merge pull request #8 from anyone-can-test/master

fix for Jenkins 1.577+.

Commit:
JENKINS-33898: Auto resolve issue created by jiratestresultreporter plugin

maplesteve committed:
Merge pull request #9 from nsharma-git/master

JENKINS-33898: Auto resolve issue created by jiratestresultreporter p…

andreituicu committed:
Add autoresolve issues functionality.

andreituicu committed:
Import 2.0 code.

andreituicu committed:
Update README with the new Autoresolve issues functionality.

andreituicu committed:
Update pom.xml

andreituicu committed:
Change job config img.

catalinluta committed:
Added files via upload

catalinluta committed:
Added Implementation details

andreituicu committed:
Fix dependencies so that the Injected tests pass.

andreituicu committed:
Fix some bugs found by FindBugs maven plugin, and disable build failure for it.

andreituicu committed:
Merge branch 'master' of github.com:jenkinsci/JiraTestResultReporter-plugin

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.0

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
Fix JENKINS-34806: Cause GitHub is case insesitive when it comes to file names.

andreituicu committed:
Fix minor UI problem.

andreituicu committed:
Workaround for changes made by Jenkins in UI for selects.

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.1

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
Fix JENKINS-34904

andreituicu committed:
Add Issue Summary as tooltip.

andreituicu committed:
Some more logging.

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.2

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
Show linked issues even when the test is passed.

andreituicu committed:
Endpoint for viewing test to issue mappings.

andreituicu committed:
Add option for configuring user fields in the job configuration

andreituicu committed:
Update parent pom.

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.3

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
Update README with guidelines for issue reports and contributions.

andreituicu committed:
Update README.md

andreituicu committed:
Update README.md

catalinluta committed:
Update README.md

catalinluta committed:
Added new line to separate Setup Workflow

andreituicu committed:
JENKINS-40520 - Breaks Save button on Manage Jenkins/Configure System.

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.4

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
JENKINS-44691 - Correlates a new test failure with a resolved jira issue

andreituicu committed:
JENKINS-47645 - java.lang.NoSuchMethodError on "Configure System" page / unable to edit configuration

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.5

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

andreituicu committed:
JENKINS-47645 - java.lang.NoSuchMethodError on "Configure System" page / unable to edit configuration
* Second attempt

andreituicu committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.6

andreituicu committed:
[maven-release-plugin] prepare for next development iteration

Commit:
JENKINS-52338 - added a missing data publisher for AutoUnlink checkbox

Commit:
JENKINS-52341 - added logic to handle project residing under sub-folder

andreituicu committed:
Merge pull request #11 from krupaldesai/issue/JENKINS-52338

JENKINS-52338 - added a missing data publisher for AutoUnlink checkbox

andreituicu committed:
Merge pull request #12 from krupaldesai/issue/JENKINS-52341

JENKINS-52341 - added logic to handle project residing under sub-folder

usmonster committed:
JENKINS-51243 - Use JiraTestReporter plugin in Jenkins pipeline(Groovy) type of job (#13)

andreituicu committed:
Update README.md

Commit:
Added jenkins identifier for UserFields

I was not able via the jenkins pipeline to add the assignee. This as
the UserField option was not marked identifiable. This commit adds
the pertinent annotation to fix that.

daniel-beck-bot committed:
Use HTTPS URLs in pom.xml

zztalker committed:
Bump dependency to jira-rest-java-client-core:5.2.1

 *  Promise has moved to new namespace: io.atlassian.util.concurrent.Promise
 *  Bump dependecies for Jenkins and jenkins-ci to run tests

zztalker committed:
Merge pull request #16 from zztalker/issue/JENKINS-61171

JENKINS-61171: Bump dependency to jira-rest-java-client-core:5.2.1

zztalker committed:
add Jenkinsfile

zztalker committed:
Merge pull request #17 from zztalker/jenkinspipeline

add Jenkinsfile

zztalker committed:
Incrementalified.

zztalker committed:
Merge pull request #18 from zztalker/incrementals

Incrementalified.

zztalker committed:
prepare release JiraTestResultReporter-2.0.8

zztalker committed:
change incrementals changelist format

zztalker committed:
test

zztalker committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.8

zztalker committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.8-1

zztalker committed:
Merge remote-tracking branch 'upstream/master'

zztalker committed:
[maven-release-plugin] prepare for next development iteration

zztalker committed:
fix dependency issue with com.google.guava MoreObjects

zztalker committed:
Merge pull request #19 from zztalker/fix_dependency_issue

fix dependency issue with com.google.guava MoreObjects

zztalker committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.9

zztalker committed:
[maven-release-plugin] prepare for next development iteration

zztalker committed:
Merge pull request #15 from daniel-beck-bot/https-urls-pom

Use HTTPS URLs in pom.xml

imonteroperez committed:
Avoid duplicated entries from Jira server

imonteroperez committed:
Refactoring and some feedback from PR review

imonteroperez committed:
Improve javadoc

imonteroperez committed:
Use resolution field instead of status

imonteroperez committed:
Merge pull request #20 from imonteroperez/avoid-duplication

Hackoctoberfest - Avoid duplicated entries from Jira server

imonteroperez committed:
Merge pull request #14 from butterlyextreme/master

Added jenkins identifier for UserFields

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.10

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

basil committed:
[JENKINS-66949] JiraTestResultReporter:ERROR Unknown: java.lang.NoSuchMethodError: com.google.common.base.Throwables.throwIfUnchecked

imonteroperez committed:
New variable TEST_PACKAGE_CLASS_METHOD_NAME

imonteroperez committed:
[JENKINS-56371] Fix on multiple junit publishers

imonteroperez committed:
Merge pull request #22 from imonteroperez/display-test

New variable TEST_PACKAGE_CLASS_METHOD_NAME

imonteroperez committed:
Merge pull request #23 from imonteroperez/JENKINS-56371

[JENKINS-56371] Fix on multiple junit publishers

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.11

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

basil committed:
Merge remote-tracking branch 'origin/master' into JENKINS-66949

imonteroperez committed:
[JENKINS-56371] Fix on multiple junit publishers (2nd round)

imonteroperez committed:
Remove missed no-sense condition

imonteroperez committed:
Merge pull request #24 from imonteroperez/review-duplicated

[JENKINS-56371] Fix on multiple junit publishers (2nd round)

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.12

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

basil committed:
Fix SCM URL

imonteroperez committed:
[JENKINS-66092] Upload attachments on newly created issues

imonteroperez committed:
Merge pull request #25 from imonteroperez/attachments-support

[JENKINS-66092] Upload attachments on newly created issues

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.13

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

imonteroperez committed:
Deal with interrumpted connections

imonteroperez committed:
Merge pull request #26 from jenkinsci/bugfix-jira

Deal with interrumpted network connections with the Jira server

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.14

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

imonteroperez committed:
Fix on creating Jira issues for parameterized tests

imonteroperez committed:
Merge pull request #27 from imonteroperez/fix-parameterized-test

Fix on creating Jira issues for parameterized tests

imonteroperez committed:
[maven-release-plugin] prepare release JiraTestResultReporter-2.0.15

imonteroperez committed:
[maven-release-plugin] prepare for next development iteration

NotMyFault committed:
chore: Prepare for icon removal from core

imonteroperez committed:
Merge pull request #21 from basil/JENKINS-66949

[JENKINS-66949] JiraTestResultReporter:ERROR Unknown: java.lang.NoSuchMethodError: com.google.common.base.Throwables.throwIfUnchecked

imonteroperez committed:
Merge pull request #28 from NotMyFault/chore/master/prep-icon-removal-from-core

Prepare for icon removal from core

imonteroperez committed:
Enable CD

imonteroperez committed:
Apply suggestions from code review

Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

imonteroperez committed:
Merge pull request #29 from jenkinsci/enable-cd

Enable CD

imonteroperez committed:
Bugfix test reference on multibranch pipelines

imonteroperez committed:
Perform search on Jira server to retrieve the issue keys to autoresolve

imonteroperez committed:
Merge pull request #35 from imonteroperez/resolve-branches

Bugfix autoresolve test reference on different stage name

imonteroperez committed:
[SECURITY-2236]

imonteroperez committed:
Manage escape JQL based on JRASERVER-25092 and provide support for parameterized tests

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraUtils.java

Co-authored-by: Cathy Chan <dr.cathy.chan@gmail.com>

imonteroperez committed:
Merge pull request #54 from imonteroperez/review-duplicated-params

Manage escape JQL based on JRASERVER-25092 and provide support for parameterized tests

imonteroperez committed:
Manual creation of Jira issues should use global config only

imonteroperez committed:
Use enum JiraIssueTrigger

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraIssueTrigger.java

Co-authored-by: Cathy Chan <dr.cathy.chan@gmail.com>

imonteroperez committed:
Merge pull request #72 from imonteroperez/ui-creation-config

Manual creation of Jira issues should use global config only

Commit:
include override option to create new tickets if cached ones are already resolved

Commit:
update java version to allow the plugin can be compiled

imonteroperez committed:
Merge pull request #81 from bmunozm/bugfix/update-java-version

update java version to allow the plugin can be compiled

imonteroperez committed:
Merge pull request #80 from bmunozm/feature/overrideResolvedIssues

Include overrideResolvedIssues parameter to create new tickets if test failure are linked to already resolved ones

jetersen committed:
chore: use jenkins infra maven cd reusable workflow

Signed-off-by: Joseph Petersen <josephp90@gmail.com>

Commit:
master: add JiraTestDataTest

Signed-off-by: Ticketedmoon <shane.creedon3@mail.dcu.ie>

imonteroperez committed:
Merge pull request #102 from Ticketedmoon/feature/add-JiraTestData-test-suite

Add test class for JiraTestData.java (2 tests)

dependabot[bot] committed:
Bump actions/checkout from 2.4.0 to 3.3.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 2.4.0 to 3.3.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2.4.0...v3.3.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump gson from 2.8.9 to 2.10.1

Bumps [gson](https://github.com/google/gson) from 2.8.9 to 2.10.1.
- [Release notes](https://github.com/google/gson/releases)
- [Changelog](https://github.com/google/gson/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/gson/compare/gson-parent-2.8.9...gson-parent-2.10.1)

---
updated-dependencies:
- dependency-name: com.google.code.gson:gson
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

imonteroperez committed:
Merge pull request #120 from jenkinsci/dependabot/maven/com.google.code.gson-gson-2.10.1

Bump gson from 2.8.9 to 2.10.1

imonteroperez committed:
Merge pull request #119 from jenkinsci/dependabot/github_actions/actions/checkout-3.3.0

Bump actions/checkout from 2.4.0 to 3.3.0

imonteroperez committed:
Add ? as supported char for queries against JIRA

imonteroperez committed:
Merge pull request #133 from imonteroperez/escape-parameter-char

Add ? as supported char for queries against JIRA

imonteroperez committed:
Fix race condition on using multiple publishers

imonteroperez committed:
Merge pull request #139 from imonteroperez/fix-datapublisher-issue

Fix race condition on using multiple publishers

imonteroperez committed:
Enforce a new release

imonteroperez committed:
Restore should_release

imonteroperez committed:
Test data should be added by the JUnitResultArchiver

imonteroperez committed:
Merge pull request #141 from imonteroperez/fix-duplicated-issue

Test data should be added by the JUnitResultArchiver

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Fix on parallel branches bug

imonteroperez committed:
Feedback review

imonteroperez committed:
Merge pull request #142 from imonteroperez/parallel-bugfix

Fix on parallel branches bug

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Add attachments from the junit-attachments plugin into Jira attachments

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraTestDataPublisher.java

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraUtils.java

Co-authored-by: Cathy Chan <dr.cathy.chan@gmail.com>

imonteroperez committed:
Configure the inclusion of attachments via EnvVars

imonteroperez committed:
Incorporate a description of the env var in README

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraTestDataPublisher.java

imonteroperez committed:
Use system.getenv instead of envvars

imonteroperez committed:
Return to use envvars

imonteroperez committed:
Use databound setter instead of env var

imonteroperez committed:
Use DataBoundSetter for additionalAttachments

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraTestDataPublisher.java

Co-authored-by: Jesse Glick <jglick@cloudbees.com>

imonteroperez committed:
Update src/main/java/org/jenkinsci/plugins/JiraTestResultReporter/JiraTestDataPublisher.java

Co-authored-by: Jesse Glick <jglick@cloudbees.com>

imonteroperez committed:
Add getter

imonteroperez committed:
Merge pull request #144 from imonteroperez/additional-attachments

Add attachments from the junit-attachments plugin into Jira attachments

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Update cd.yaml

strangelookingnerd committed:
Enable Jenkins Security Scan

imonteroperez committed:
Merge pull request #146 from strangelookingnerd/patch-1

Enable Jenkins Security Scan

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Bump BOM fo fix JUnit issues

imonteroperez committed:
Fix enforcer rules

imonteroperez committed:
Temporary ignore spotbugs issues

imonteroperez committed:
Merge pull request #147 from imonteroperez/junit-fix

Bump BOM fo fix JUnit issues

imonteroperez committed:
Update cd.yaml

imonteroperez committed:
Merge branch 'master' into chore/maven-cd-reusable-workflow

imonteroperez committed:
Merge pull request #83 from jetersen-cloud/chore/maven-cd-reusable-workflow

chore: use jenkins infra maven cd reusable workflow

dependabot[bot] committed:
Bump io.jenkins.tools.incrementals:git-changelist-maven-extension (#149)

Bumps [io.jenkins.tools.incrementals:git-changelist-maven-extension](https://github.com/jenkinsci/incrementals-tools) from 1.2 to 1.8.
- [Release notes](https://github.com/jenkinsci/incrementals-tools/releases)
- [Commits](https://github.com/jenkinsci/incrementals-tools/compare/parent-1.2...parent-1.8)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.incrementals:git-changelist-maven-extension
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

basil committed:
Refresh plugin for September 2024

imonteroperez committed:
Merge pull request #153 from basil/refresh

Refresh plugin for September 2024

chcg committed:
- fixed issue #157 on ClassNotFoundException com.atlassian.httpclient.apache.httpcomponents.DefaultHttpClientFactory (#158)

- corrected some typos in README.md
- removed unneeded imports

basil committed:
Lower baseline to 2.452.3 (#161)

basil committed:
Format repository with Spotless (#164)

basil committed:
Add .git-blame-ignore-revs (#165)

dependabot[bot] committed:
Bump jira-rest-client.version from 5.2.6 to 6.0.1

Bumps `jira-rest-client.version` from 5.2.6 to 6.0.1.

Updates `com.atlassian.jira:jira-rest-java-client-api` from 5.2.6 to 6.0.1
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.1..jira-rest-java-client-parent-5.2.6)

Updates `com.atlassian.jira:jira-rest-java-client-core` from 5.2.6 to 6.0.1
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.1..jira-rest-java-client-parent-5.2.6)

---
updated-dependencies:
- dependency-name: com.atlassian.jira:jira-rest-java-client-api
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: com.atlassian.jira:jira-rest-java-client-core
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

chcg committed:
Add bearer token support (#159)

jglick committed:
NPE fix from `JiraTestDataPublisher.getJobName`

jglick committed:
Spotless

imonteroperez committed:
Merge pull request #168 from jglick/NPE

NPE fix from `JiraTestDataPublisher.getJobName`

chcg committed:
Fix Java compiler warning (#171)

Co-authored-by: Basil Crow <me@basilcrow.com>

basil committed:
Test on Java 21 (#180)

chcg committed:
Replace deprecated `getInstance` and `getActiveInstance` calls with non-deprecated equivalents (#179)

chcg committed:
Auto unlink documentation (#178)

yaroslavafenkin committed:
[JENKINS-74234] Remove unused jelly view (#182)

yaroslavafenkin committed:
[JENKINS-74233] Extract inline JavaScript from `JiraTestDataPublisher/config.jelly` (#181)

chcg committed:
allow update to 3654.v237e4a_f2d8da_ as seen in PR #175 needs jenkins.version 2.452.4 (#184)

yaroslavafenkin committed:
[JENKINS-74232][JENKINS-74235] Extract inline JavaScript (#183)

chcg committed:
- adapted header for consistency
- added code markdown for command line calls
- added description for images

chcg committed:
- added nullpointer checks
- consolidation of usage @NonNull vs. @Nonnull

imonteroperez committed:
Merge pull request #191 from chcg/issue_189

Updated docu

chcg committed:
fix #193
- add missing parts for feature overrideResolvedIssues from PR #80

imonteroperez committed:
Merge pull request #194 from chcg/issue_193

enable overrideResolvedIssues

chcg committed:
Merge pull request #192 from chcg/issue_190

avoid some NPE on generateSnippet

chcg commented:
@lillysGame Active development of the current plugin version is at https://github.com/jenkinsci/JiraTestResultReporter-plugin/.

basil committed:
Migrate from EE 8 to EE 9 (#207)

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 3850.vb_c5319efa_e29 to 4051.v78dce3ce8b_d6.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

chcg committed:
- update to jira-rest-client version 5.2.8-platform7-12284590
- update baseline version to .3
- fix typo
- add code comment

chcg committed:
Merge pull request #210 from chcg/pom_updates

jira-rest-client version 5.2.8-platform7-12284590

chcg committed:
Merge branch 'master' into dependabot/maven/jira-rest-client.version-6.0.1

chcg committed:
Merge pull request #208 from jenkinsci/dependabot/maven/io.jenkins.tools.bom-bom-2.479.x-4051.v78dce3ce8b_d6

Bump io.jenkins.tools.bom:bom-2.479.x from 3850.vb_c5319efa_e29 to 4051.v78dce3ce8b_d6

chcg committed:
avoid NPE warning in unused code by VS Code Java

chcg committed:
- fix VS Code java warning about Job is a raw type. References to generic type Job<E> should be parameterized

chcg committed:
fix spotless findings

chcg committed:
Merge pull request #212 from chcg/avoid_raw

Avoid raw type warnings

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4051.v78dce3ce8b_d6 to 4136.vca_c3202a_7fd1.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

chcg committed:
Merge pull request #213 from jenkinsci/dependabot/maven/io.jenkins.tools.bom-bom-2.479.x-4136.vca_c3202a_7fd1

Bump io.jenkins.tools.bom:bom-2.479.x from 4051.v78dce3ce8b_d6 to 4136.vca_c3202a_7fd1

chcg committed:
Merge pull request #152 from jenkinsci/dependabot/maven/jira-rest-client.version-6.0.1

Bump jira-rest-client.version from 5.2.8 to 6.0.1

dependabot[bot] committed:
Bump io.atlassian.fugue:fugue from 4.7.2 to 6.1.2

Bumps [io.atlassian.fugue:fugue](https://bitbucket.org/atlassian/fugue) from 4.7.2 to 6.1.2.
- [Changelog](https://bitbucket.org/atlassian/fugue/src/master/changelog.md)
- [Commits](https://bitbucket.org/atlassian/fugue/branches/compare/v6.1.2..v4.7.2)

---
updated-dependencies:
- dependency-name: io.atlassian.fugue:fugue
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

chcg committed:
Merge pull request #209 from jenkinsci/dependabot/maven/io.atlassian.fugue-fugue-6.1.2

Bump io.atlassian.fugue:fugue from 4.7.2 to 6.1.2

chcg committed:
validateFieldConfigs json changed (#214)

* fix typo

* seems json structure changed for publisher to JSONArray, adapted code to support that in case object is not found

* revert

Bump io.atlassian.fugue:fugue from 4.7.2 to 6.1.2 (#209) @dependabot[bot]
Bump jira-rest-client.version from 5.2.6 to 6.0.1 (#152) @dependabot[bot]

as they start to cause build issues

* fix spotless issues

dependabot[bot] committed:
Bump io.atlassian.fugue:fugue from 4.7.2 to 6.1.2 (#216)

Bumps [io.atlassian.fugue:fugue](https://bitbucket.org/atlassian/fugue) from 4.7.2 to 6.1.2.
- [Changelog](https://bitbucket.org/atlassian/fugue/src/master/changelog.md)
- [Commits](https://bitbucket.org/atlassian/fugue/branches/compare/v6.1.2..v4.7.2)

---
updated-dependencies:
- dependency-name: io.atlassian.fugue:fugue
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jira-rest-client.version from 5.2.8-platform7-12284590 to 6.0.1 (#215)

Bumps `jira-rest-client.version` from 5.2.8-platform7-12284590 to 6.0.1.

Updates `com.atlassian.jira:jira-rest-java-client-api` from 5.2.8-platform7-12284590 to 6.0.1
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.1..jira-rest-java-client-parent-5.2.8-platform7-12284590)

Updates `com.atlassian.jira:jira-rest-java-client-core` from 5.2.8-platform7-12284590 to 6.0.1
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.1..jira-rest-java-client-parent-5.2.8-platform7-12284590)

---
updated-dependencies:
- dependency-name: com.atlassian.jira:jira-rest-java-client-api
  dependency-type: direct:production
  update-type: version-update:semver-major
- dependency-name: com.atlassian.jira:jira-rest-java-client-core
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: chcg <christian.grasser@web.de>

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x (#217)

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4136.vca_c3202a_7fd1 to 4228.v0a_71308d905b_.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chcg committed:
JENKINS-69312 JiraTestResultReporter plugin does not show plus button to create an issue (#211)

* - fix for https://issues.jenkins.io/browse/JENKINS-69312 and dup https://issues.jenkins.io/browse/JENKINS-67777
  with an additional checkbox to enable manual linking

* avoid potentially problematic resorting of members, see https://github.com/jenkinsci/JiraTestResultReporter-plugin/pull/211#pullrequestreview-2636255936

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:plugin from 5.7 to 5.9 (#220)

Bumps [org.jenkins-ci.plugins:plugin](https://github.com/jenkinsci/plugin-pom) from 5.7 to 5.9.
- [Release notes](https://github.com/jenkinsci/plugin-pom/releases)
- [Changelog](https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/plugin-pom/compare/plugin-5.7...plugin-5.9)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:plugin
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump jira-rest-client.version from 6.0.1 to 6.0.2 (#219)

Bumps `jira-rest-client.version` from 6.0.1 to 6.0.2.

Updates `com.atlassian.jira:jira-rest-java-client-api` from 6.0.1 to 6.0.2
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.2..jira-rest-java-client-parent-6.0.1)

Updates `com.atlassian.jira:jira-rest-java-client-core` from 6.0.1 to 6.0.2
- [Commits](https://bitbucket.org/atlassian/jira-rest-java-client/branches/compare/jira-rest-java-client-parent-6.0.2..jira-rest-java-client-parent-6.0.1)

---
updated-dependencies:
- dependency-name: com.atlassian.jira:jira-rest-java-client-api
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: com.atlassian.jira:jira-rest-java-client-core
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chcg committed:
try to fix some issues with generateSnippets for pipeline configs to get at least some partial output from snippet generator (#221)

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x (#222)

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4228.v0a_71308d905b_ to 4488.v7fe26526366e.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chcg committed:
Better Log for Issue #218 (#223)

* fix some NPE issues from vs code also catched via try/catch

* avoid exception trace

RestClientException{statusCode=Optional.of(404), errorCollections=[ErrorCollection{status=404, errors={}, errorMessages=[Issue Does Not Exist]}]}
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:176)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:170)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:49)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:10)
	at PluginClassLoader for JiraTestResultReporter//io.atlassian.util.concurrent.Promises$OfStage.lambda$fold$4(Promises.java:332)
	at PluginClassLoader for JiraTestResultReporter//io.atlassian.util.concurrent.Promises.lambda$biFunction$7(Promises.java:422)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$1.lambda$doCompleted$0(SettableFuturePromiseHttpPromiseAsyncClient.java:39)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient.runInContext(SettableFuturePromiseHttpPromiseAsyncClient.java:80)
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$ThreadLocalDelegateRunnable.run(SettableFuturePromiseHttpPromiseAsyncClient.java:148)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
Caused: RestClientException{statusCode=Optional.of(404), errorCollections=[ErrorCollection{status=404, errors={}, errorMessages=[Issue Does Not Exist]}]}
	at PluginClassLoader for JiraTestResultReporter//com.atlassian.jira.rest.client.internal.async.DelegatingPromise.claim(DelegatingPromise.java:45)
	at PluginClassLoader for JiraTestResultReporter//org.jenkinsci.plugins.JiraTestResultReporter.MetadataCache.getCacheEntry(MetadataCache.java:184)
	at PluginClassLoader for JiraTestResultReporter//org.jenkinsci.plugins.JiraTestResultReporter.JiraTestDataPublisher$JiraTestDataPublisherDescriptor.getCacheEntry(JiraTestDataPublisher.java:564)
	at PluginClassLoader for JiraTestResultReporter//org.jenkinsci.plugins.JiraTestResultReporter.config.StringFields$StringFieldsDescriptor.doFillFieldKeyItems(StringFields.java:124)

---------

Co-authored-by: grasser <christian.grasser@elektrobit.com>

imonteroperez committed:
Fix junit-attachments incompatibility

imonteroperez committed:
Fix spotless

imonteroperez committed:
Merge pull request #226 from jenkinsci/junit-attachments-compatibility

Fix junit-attachments incompatibility

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x (#228)

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4545.v56392b_7ca_7b_a_ to 4570.v1b_c718dd3b_1e.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-version: 4570.v1b_c718dd3b_1e
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4570.v1b_c718dd3b_1e to 4607.v67a_4791074d7.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-version: 4607.v67a_4791074d7
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:plugin from 5.9 to 5.10

Bumps [org.jenkins-ci.plugins:plugin](https://github.com/jenkinsci/plugin-pom) from 5.9 to 5.10.
- [Release notes](https://github.com/jenkinsci/plugin-pom/releases)
- [Changelog](https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/plugin-pom/compare/plugin-5.9...plugin-5.10)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:plugin
  dependency-version: '5.10'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

imonteroperez committed:
Merge pull request #229 from jenkinsci/dependabot/maven/io.jenkins.tools.bom-bom-2.479.x-4607.v67a_4791074d7

Bump io.jenkins.tools.bom:bom-2.479.x from 4570.v1b_c718dd3b_1e to 4607.v67a_4791074d7

imonteroperez committed:
Merge pull request #230 from jenkinsci/dependabot/maven/org.jenkins-ci.plugins-plugin-5.10

Bump org.jenkins-ci.plugins:plugin from 5.9 to 5.10

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:plugin from 5.10 to 5.11 (#231)

Bumps [org.jenkins-ci.plugins:plugin](https://github.com/jenkinsci/plugin-pom) from 5.10 to 5.11.
- [Release notes](https://github.com/jenkinsci/plugin-pom/releases)
- [Changelog](https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/plugin-pom/compare/plugin-5.10...plugin-5.11)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:plugin
  dependency-version: '5.11'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x (#232)

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4607.v67a_4791074d7 to 4628.v2b_234a_1a_20d0.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-version: 4628.v2b_234a_1a_20d0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:plugin from 5.11 to 5.12

Bumps [org.jenkins-ci.plugins:plugin](https://github.com/jenkinsci/plugin-pom) from 5.11 to 5.12.
- [Release notes](https://github.com/jenkinsci/plugin-pom/releases)
- [Changelog](https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/plugin-pom/compare/plugin-5.11...plugin-5.12)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:plugin
  dependency-version: '5.12'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

imonteroperez committed:
Merge pull request #234 from jenkinsci/dependabot/maven/org.jenkins-ci.plugins-plugin-5.12

Bump org.jenkins-ci.plugins:plugin from 5.11 to 5.12

maplesteve commented:
@chcg, @imonteroperez This repo was for the initial version of the plugin which I created years ago. The maintained version lives here: https://github.com/jenkinsci/JiraTestResultReporter-plugin
I'll archive this repo in coming days to avoid further confusion. So, if there's anything relevant in this PR, please transfer it to a new PR in the official repo.

dependabot[bot] committed:
Bump io.jenkins.tools.bom:bom-2.479.x (#233)

Bumps [io.jenkins.tools.bom:bom-2.479.x](https://github.com/jenkinsci/bom) from 4628.v2b_234a_1a_20d0 to 4669.v0e99c712a_30e.
- [Release notes](https://github.com/jenkinsci/bom/releases)
- [Commits](https://github.com/jenkinsci/bom/commits)

---
updated-dependencies:
- dependency-name: io.jenkins.tools.bom:bom-2.479.x
  dependency-version: 4669.v0e99c712a_30e
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ildefonso Montero <ildefonso.montero@gmail.com>

chcg commented:
@maplesteve The changes here are reflecting the changes within https://github.com/jenkinsci/JiraTestResultReporter-plugin, so nothing relevant is within this PR and you could archive your repo.

maplesteve commented:
This repo was for the initial version of the plugin which I created years ago. The maintained version lives here: https://github.com/jenkinsci/JiraTestResultReporter-plugin
